x-pack/plugins/triggers_actions_ui/public/application/lib/rule_api/state.ts

```diff
@@ -33,17 +29,9 @@ export async function loadRuleState({
   http: HttpSetup;
   ruleId: string;
 }): Promise<RuleTaskState> {
-  return await http
+  return (await http
     .get<AsApiContract<RuleTaskState> | EmptyHttpResponse>(
       `${INTERNAL_BASE_ALERTING_API_PATH}/rule/${ruleId}/state`
     )
-    .then((state) => (state ? rewriteBodyRes(state) : {}))
-    .then((state: RuleTaskState) => {
-      return pipe(
-        ruleStateSchema.decode(state),
-        fold((e: Errors) => {
-          throw new Error(`Rule "${ruleId}" has invalid state`);
-        }, identity)
-      );
-    });
```

Comment:
No longer necessary because:
- The validation happens on task save
- The decode would convert string dates into date objects. But the response handle would convert them back into strings..